docs/what-it-does.md

```diff
@@ -0,0 +1,103 @@
+# What the Consul Deployment Agent Does
+
+## On startup
+
+- Waits for puppet to drop file to signal instance readiness before proceeding.
```

Comment:
If puppet is dropping a configuration file that *does* wait for this, otherwise not. So while this is the case at the moment, it is optional.